### DISCUSSION TITLE
2022.7.0

### DISCUSSION THREAD

**Discussion by Thomas55555** [Thomas55555/husqvarna_automower#276] [Announcements]
![downloads](https://img.shields.io/github/downloads/Thomas55555/husqvarna_automower/2022.7.0/total.svg)
## What’s Changed

* remove token_info from entry (#275) @Thomas55555
* Add Binary Error sensor (#271) @Thomas55555
* Relinked the screenshot in the README file. (#273) @pjbaert
* add mower name to park and start entity (#272) @Thomas55555
* bump HA reqs to 2022.7.0b0 (#270) @Thomas55555
* temporarily downgrade hacs reqs (#269) @Thomas55555
* make percentage_sensors less boring (#268) @Thomas55555
* number_entity for park and start (#267) @Thomas55555
* Migrate number entity to native value (#257) @Thomas55555

## Breaking changes / Deprecations
* remove log-in with password/username (#274) @Thomas55555
  Please add your client_id and client_secret to the `configuration.yaml`: https://github.com/Thomas55555/husqvarna_automower#home-assistant
* The service `park and start` is depracated. You can use the new `number_entities`(#272) instead. Please update your automations....

<hr /><em>This discussion was created from the release <a href='https://github.com/Thomas55555/husqvarna_automower/releases/tag/2022.7.0'>2022.7.0</a>.</em>

**Comment by slovdahl** [comment]
The breaking changes lists this entry:

> Please add your client_id and client_secret to the configuration.yaml

But there is no documentation on how to add `client_id` and `client_secret` to `configuration.yaml` anywhere 🤔 What am I missing?

**Comment by Thomas55555** [comment]
Sorry, I've changed the documentation already, because I'm planning to enter this UI. But here is the old documentation:
https://github.com/Thomas55555/husqvarna_automower/tree/2022.7.0#home-assistant

**Comment by dawiinci** [comment]
So, what do I have to do now? It tells me the configuration with the api key is deprecated. When I remove it, it still works. But I also get this:
`Log-in with password/username is depracated. Please set-up client_id and client_secret in the Application Credentials form`

There is no configuration in the configuration pane of the plugin - only the camera settings.

What is this "Application Credentials form"?

**Comment by Thomas55555** [comment]
There is the Application Credentials form:
https://my.home-assistant.io/redirect/application_credentials/
The warning should not show, if you've removed the key from the `configuration.yaml`. There I did something wrong. :(

**Comment by dawiinci** [comment]
Well I now have the request to login again under devices. When I do that the website opens and I can login. After that I get this error from husqvarna:
`Invalid client: `redirect_uri` does not match client value
`
My redirect url looks like that:
`localhost:8123/redirect/oauth`

Is there something wrong?

**Comment by dawiinci** [comment]
Now that I deleted my API key and made a new one, I can't even try to login:

```
That was an error...
Client not found
```
I only have the api and secret key inside of secrets.yaml. Nothing in the configuration.yaml. I restarted HA.

Why is this not working??

**Reply by slovdahl** [reply]
> I only have the api and secret key inside of secrets.yaml. Nothing in the configuration.yaml. I restarted HA.

As far as I understand it, just adding something to `secrets.yaml` won't automagically make it be used anywhere. You probably need to add this to your `configuration.yaml` too:

```yaml
husqvarna_automower:
  client_id: !secret husqvarna_apikey
  client_secret: !secret husqvarna_client_secret
```

**Reply by dawiinci** [reply]
Thanks it worked. But I also had to enable the my integration in the configuration.yaml (my:)

**Reply by Thomas55555** [reply]
Suggestion:
- Upgrade to husqvarna_automower 2022.7.1. there you don't have to put anything in the `secrets.yaml` or `configuration.yaml`
- Delete your entries in the Application Cedentials form
- Delete the integration and watch this video, how to setup the integration again: https://user-images.githubusercontent.com/59625598/178572694-73986b4d-de13-4950-a774-be99da21defa.mp4

These seemd to help a lot of people here: #296

**Comment by dawiinci** [comment]
I deleted the device and still get the same error. Now I can't even set up the plugin.
```
That was an error...
Client not found
```

It still uses the old `client_id`. How can I completely reset the plugin? Where is this information stored?

**Reply by Thomas55555** [reply]
It's stored in the Application credentials form. Delete it there.
